### PR TITLE
1st version of membership flow

### PR DESCRIPTION
**Pull request by pritamkarmakar**
Major Changes:
1. Membership page for already logged-in user and not logged-in user.
2. Membership details page (in-progress)
3. code refactoring 

Demo:
membership_flow

### CODE REVIEW

**Review by prarthanaK — COMMENTED** [review]
Left a few thoughts/suggestions.

File: lib/core/models/event/event_static_data.dart [hunk 1]
```diff
@@ -80,6 +88,8 @@ class EventStaticData {
           'Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum ',
       startDate: DateTime(2021, 04, 21),
       endDate: DateTime(2021, 04, 21),
+      eventBannerImagePath:
+          'https://firebasestorage.googleapis.com/v0/b/bhindeshi-app.appspot.com/o/event_banner%2F5.jpg?alt=media&token=433c8688-3a10-48c2-bae1-fae9249b6d8e',
```
[the remark this answers is not in the document]
Comment (pritamkarmakar) [reply lib/core/models/event/event_static_data.dart:92]: We're using this static content to push this to our database. Only admins will have access to generate the events and this is a one-time operation. Then if we need to change anything we will do it directly in the database.

File: lib/core/models/event/event_static_data.dart [hunk 2]
```diff
@@ -92,6 +102,8 @@ class EventStaticData {
       startDate: DateTime(2021, 04, 21),
       endDate: DateTime(2021, 04, 21),
       location: 'Zoom meeting',
+      eventBannerImagePath:
+          'https://firebasestorage.googleapis.com/v0/b/bhindeshi-app.appspot.com/o/event_banner%2F4.jpg?alt=media&token=78d5d3df-76fd-4471-bc49-8f2d86852d26',
```
[the remark this answers is not in the document]
Comment (pritamkarmakar) [reply lib/core/models/event/event_static_data.dart:106]: Please see above comment.

File: lib/core/models/membership/user_membership_info.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class MembershipTransaction {
+  final String transactionId;
+  final String membershipId;
+  final DateTime transactionDate;
+  final double membershipCost;
```
[the remark this answers is not in the document]
Comment (pritamkarmakar) [reply lib/core/models/membership/user_membership_info.dart:5]: Good suggestion, I have added currency in MembershipInfo data model. Found that we don't need this data model so deleting it.

File: lib/core/services/constants/database_collections_name.dart [hunk 4]
```diff
@@ -0,0 +1,4 @@
+const USER_ROLE_COLLECTION = 'user_role';
```
[the remark this answers is not in the document]
Comment (pritamkarmakar) [reply lib/core/services/constants/database_collections_name.dart:1]: done.

File: lib/core/services/constants/database_collections_name.dart [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+const MEMBERSHIP_COLLECTION = 'membership';
+const USER_MEMBERSHIP_COLLECTION = 'users_membership_info';
```
[the remark this answers is not in the document]
Comment (pritamkarmakar) [reply lib/core/services/constants/database_collections_name.dart:3]: Fixed.

**Comment by pritamkarmakar** [comment]
https://github.com/pritamkarmakar/bhindeshi_app/pull/3 has all these changes so we don't need this.

**Review by prarthanaK — COMMENTED** [review]
LGTM

**Pull request closed without merging**